modules/extension/wmts/src/main/java/org/geotools/ows/wmts/map/WMTSCoverageReader.java

```diff
@@ -563,4 +564,141 @@ public String getRequestedTime() {
     public void setRequestedTime(String requestedTime) {
         this.requestedTime = requestedTime;
     }
+
+    boolean delegateToRemote(CoordinateReferenceSystem crs) {
+        String srs = CRS.toSRS(crs);
+        return srs != null && validSRS.contains(srs);
+    }
+
+    private String switchDefinition(CoordinateReferenceSystem crs) {
+        String srs = CRS.toSRS(crs);
+        String result;
+        if (!srs.startsWith("urn:ogc:def:crs:EPSG::"))
+            result = srs.replace("EPSG:", "urn:ogc:def:crs:EPSG::");
+        else result = srs.replace("urn:ogc:def:crs:EPSG::", "EPSG");
+        if (!validSRS.contains(result)) result = switchPseudoMercatorDefinition(srs);
+        if (result != null && validSRS.contains(result)) return result;
+        else return null;
+    }
+
+    private String switchPseudoMercatorDefinition(String srs) {
+        String switched = null;
+        if (srs.equals("EPSG:3857") || srs.equals("urn:ogc:def:crs:EPSG::3857")) {
+            if (validSRS.contains("EPSG:900913")) {
+                switched = "EPSG:900913";
+            } else if (validSRS.contains("urn:ogc:def:crs:EPSG::900913")) {
+                switched = "urn:ogc:def:crs:EPSG::900913";
+            }
+        }
+
+        if (srs.equals("EPSG:90013") || srs.equals("urn:ogc:def:crs:EPSG::900913")) {
+            if (validSRS.contains("EPSG:3857")) {
+                switched = "EPSG:3857";
+            } else if (validSRS.contains("urn:ogc:def:crs:EPSG::3857")) {
+                switched = "urn:ogc:def:crs:EPSG::3857";
+            }
+        }
+        return switched;
+    }
+
+    private GridCoverage2D reprojectManually(
```

Comment:
```suggestion
    private GridCoverage2D reproject(
```

No hand gets dirty here

Reply:
fixed